src/engine/source/kvdb/src/kvdbManager.cpp

```diff
@@ -1,412 +1,382 @@
-        return nullptr;
+        initializeOptions();
+        initializeMainDB();
+        m_isInitialized = true;
     }
+}
 
-    LOG_DEBUG(
-        "Engine KVDB manager: '{}' method: Loading database '{}' to the available databases list.", __func__, name);
-    auto kvdb = std::make_shared<KVDB>(name, m_dbStoragePath);
-    auto result = kvdb->init(createIfMissing, false, m_spMetricsScope);
-    if (KVDB::CreationStatus::OkInitialized == result
-        || KVDB::CreationStatus::OkCreated == result)
+void KVDBManager::finalize()
+{
+    if (m_isInitialized)
     {
-        // This instrument measures the number of KVDB created
-        createdDatabases->addValue(1UL);
-        // This instrument measures the current number of active KVDB
-        databasesInUse->addValue(1L);
-
-        m_dbs[name] = kvdb;
-        return kvdb;
+        finalizeMainDB();
+        m_isInitialized = false;
     }
+}
 
-    return nullptr;
+void KVDBManager::initializeOptions()
+{
+    m_rocksDBOptions = rocksdb::Options();
+    m_rocksDBOptions.IncreaseParallelism();
+    m_rocksDBOptions.OptimizeLevelStyleCompaction();
+    m_rocksDBOptions.create_if_missing = true;
 }
 
-void KVDBManager::unloadDB(const std::string& name)
+void KVDBManager::initializeMainDB()
 {
-    std::unique_lock lkW(m_mtx);
-    const bool isLoaded = m_dbs.find(name) != m_dbs.end();
+    const auto dbStoragePath = m_ManagerOptions.dbStoragePath.string();
+
+    std::filesystem::create_directories(dbStoragePath);
+
+    const std::string dbNameFullPath {fmt::format("{}{}", dbStoragePath, m_ManagerOptions.dbName)};
+
+    std::vector<std::string> columnNames;
 
-    auto databasesInUse = m_spMetricsScope->getUpDownCounterInteger("DatabasesInUse");
+    std::vector<rocksdb::ColumnFamilyDescriptor> cfDescriptors;
+    std::vector<rocksdb::ColumnFamilyHandle*> cfHandles;
 
-    if (isLoaded)
+    bool hasDefaultCF = false;
+    const auto listStatus = rocksdb::DB::ListColumnFamilies(rocksdb::DBOptions(), dbNameFullPath, &columnNames);
+    if (listStatus.ok())
     {
-        m_dbs.erase(name);
+        for (const auto& cfName : columnNames)
+        {
+            if (rocksdb::kDefaultColumnFamilyName == cfName)
+            {
+                hasDefaultCF = true;
+            }
 
-        // This instrument measures the current number of active KVDB
-        databasesInUse->addValue(-1L);
+            auto newDescriptor = rocksdb::ColumnFamilyDescriptor(cfName, rocksdb::ColumnFamilyOptions());
+            cfDescriptors.push_back(newDescriptor);
+        }
     }
-}
 
-KVDBHandle KVDBManager::getDB(const std::string& name)
-{
-    std::shared_lock lkReadDBs(m_mtx);
+    if (!hasDefaultCF)
+    {
+        auto newDescriptor =
+            rocksdb::ColumnFamilyDescriptor(rocksdb::kDefaultColumnFamilyName, rocksdb::ColumnFamilyOptions());
+        cfDescriptors.push_back(newDescriptor);
+    }
```

Comment:
Can this case happen ? if so, then I guess that we could force it to create kvdbs without default cf, right? if not, can we change **kDefaultColumnFamilyName** to a not-so-common name ?

Reply:
It is not possible not to load the "default" CF, since it is mandatory for backward compatibility.

If the "default" CF is not specified RocksDB throws the exception "An error occurred while trying to open the database: Invalid argument: Default column family not specified".

For more information: https://github.com/EighteenZi/rocksdb_wiki/blob/master/Column-Families.md